presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -120,24 +138,42 @@ public static CachingHiveMetastore memoizeMetastore(ExtendedHiveMetastore delega
                 newDirectExecutorService(),
                 OptionalLong.empty(),
                 OptionalLong.empty(),
-                maximumSize);
+                maximumSize,
+                false,
+                ALL);
     }
 
-    private CachingHiveMetastore(ExtendedHiveMetastore delegate, ExecutorService executor, OptionalLong expiresAfterWriteMillis, OptionalLong refreshMills, long maximumSize)
+    private CachingHiveMetastore(
+            ExtendedHiveMetastore delegate,
+            ExecutorService executor,
+            OptionalLong expiresAfterWriteMillis,
+            OptionalLong refreshMills,
+            long maximumSize,
+            boolean isPartitionVersioningEnabled,
+            MetastoreCacheScope metastoreCacheScope)
     {
         this.delegate = requireNonNull(delegate, "delegate is null");
         requireNonNull(executor, "executor is null");
+        this.isPartitionVersioningEnabled = isPartitionVersioningEnabled;
+
+        OptionalLong genExpiresAfterWriteMillis = metastoreCacheScope == ALL ? expiresAfterWriteMillis : OptionalLong.of(0);
+        OptionalLong genRefreshMills = metastoreCacheScope == ALL ? refreshMills : OptionalLong.of(0);
```

Comment:
What is `gen`?

Reply:
I meant general. Renamed to `cacheExpiresAfterWriteMillis`

Reply:
I see. We use `gen` for async php code and I guess my mind just drifts to the wrong side 😂